# PrimeTrade AI Internship Assignment

## Bitcoin Market Sentiment vs Hyperliquid Trader Performance

### Objective

The goal of this project is to analyze the relationship between Bitcoin market sentiment (Fear & Greed Index) and trader performance on the Hyperliquid platform.

This analysis aims to uncover meaningful trading patterns, identify how sentiment influences profitability, and generate actionable insights using Exploratory Data Analysis (EDA).

---

### Datasets

1. Bitcoin Fear & Greed Index
2. Hyperliquid Historical Trader Data

---

### Technologies Used

- Python
- Pandas
- NumPy
- Matplotlib
- Seaborn
- Jupyter Notebook

# 1. Import Required Libraries

In [3]:
# Data manipulation
import pandas as pd
import numpy as np

# Data visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Ignore warnings
import warnings
warnings.filterwarnings("ignore")

# Display settings
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)
pd.set_option("display.float_format", "{:.2f}".format)

# Visualization settings
plt.style.use("ggplot")
sns.set_theme(style="whitegrid")

print("Libraries imported successfully.")

Libraries imported successfully.


# 2. Load Datasets

In [6]:
# File paths

historical_path = "../data/raw/historical_data.csv"
sentiment_path = "../data/raw/fear_greed_index.csv"

# Load datasets

historical_df = pd.read_csv(historical_path)
sentiment_df = pd.read_csv(sentiment_path)

print("Datasets loaded successfully.")

Datasets loaded successfully.


# 3. Dataset Overview

In [9]:
print("="*60)
print("Historical Trader Dataset")
print("="*60)

display(historical_df.head())

print("\nShape:", historical_df.shape)

Historical Trader Dataset


,Account,Coin,Execution Price,Size Tokens,Size USD,Side,Timestamp IST,Start Position,Direction,Closed PnL,Transaction Hash,Order ID,Crossed,Fee,Trade ID,Timestamp
0,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.98,986.87,7872.16,BUY,02-12-2024 22:50,0.00,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.35,895000000000000.00,1730000000000.00
1,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.98,16.00,127.68,BUY,02-12-2024 22:50,986.52,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.01,443000000000000.00,1730000000000.00
2,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,144.09,1150.63,BUY,02-12-2024 22:50,1002.52,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.05,660000000000000.00,1730000000000.00
3,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,142.98,1142.04,BUY,02-12-2024 22:50,1146.56,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.05,1080000000000000.00,1730000000000.00
4,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,8.73,69.75,BUY,02-12-2024 22:50,1289.49,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.00,1050000000000000.00,1730000000000.00



Shape: (211224, 16)


In [11]:
print("="*60)
print("Fear & Greed Dataset")
print("="*60)

display(sentiment_df.head())

print("\nShape:", sentiment_df.shape)

Fear & Greed Dataset


,timestamp,value,classification,date
0,1517463000,30,Fear,2018-02-01
1,1517549400,15,Extreme Fear,2018-02-02
2,1517635800,40,Fear,2018-02-03
3,1517722200,24,Extreme Fear,2018-02-04
4,1517808600,11,Extreme Fear,2018-02-05



Shape: (2644, 4)


# Initial Observations

In this section we inspect both datasets to understand their structure before performing any preprocessing.

The inspection includes:

- Number of rows and columns
- Sample records
- Feature names
- Overall dataset dimensions

This helps identify potential data quality issues before cleaning.

# 4. Dataset Inspection

Before performing any preprocessing, we inspect both datasets to understand their structure and quality.

This section includes:

- Column names
- Data types
- Missing values
- Duplicate records
- Summary statistics
- Memory usage

These checks help identify potential data quality issues and guide the preprocessing steps.

In [15]:
print("=" * 70)
print("Historical Trader Dataset Information")
print("=" * 70)

historical_df.info()

Historical Trader Dataset Information
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 211224 entries, 0 to 211223
Data columns (total 16 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Account           211224 non-null  object 
 1   Coin              211224 non-null  object 
 2   Execution Price   211224 non-null  float64
 3   Size Tokens       211224 non-null  float64
 4   Size USD          211224 non-null  float64
 5   Side              211224 non-null  object 
 6   Timestamp IST     211224 non-null  object 
 7   Start Position    211224 non-null  float64
 8   Direction         211224 non-null  object 
 9   Closed PnL        211224 non-null  float64
 10  Transaction Hash  211224 non-null  object 
 11  Order ID          211224 non-null  int64  
 12  Crossed           211224 non-null  bool   
 13  Fee               211224 non-null  float64
 14  Trade ID          211224 non-null  float64
 15  Timestamp         211224 non-n

In [17]:
print("=" * 70)
print("Fear & Greed Dataset Information")
print("=" * 70)

sentiment_df.info()

Fear & Greed Dataset Information
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2644 entries, 0 to 2643
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   timestamp       2644 non-null   int64 
 1   value           2644 non-null   int64 
 2   classification  2644 non-null   object
 3   date            2644 non-null   object
dtypes: int64(2), object(2)
memory usage: 82.8+ KB


In [19]:
print("Historical Dataset Columns")
print(historical_df.columns.tolist())

print("\nFear & Greed Dataset Columns")
print(sentiment_df.columns.tolist())

Historical Dataset Columns
['Account', 'Coin', 'Execution Price', 'Size Tokens', 'Size USD', 'Side', 'Timestamp IST', 'Start Position', 'Direction', 'Closed PnL', 'Transaction Hash', 'Order ID', 'Crossed', 'Fee', 'Trade ID', 'Timestamp']

Fear & Greed Dataset Columns
['timestamp', 'value', 'classification', 'date']


In [21]:
print("=" * 70)
print("Missing Values")
print("=" * 70)

print("\nHistorical Dataset")
display(historical_df.isnull().sum())

print("\nFear & Greed Dataset")
display(sentiment_df.isnull().sum())

Missing Values

Historical Dataset


Account             0
Coin                0
Execution Price     0
Size Tokens         0
Size USD            0
Side                0
Timestamp IST       0
Start Position      0
Direction           0
Closed PnL          0
Transaction Hash    0
Order ID            0
Crossed             0
Fee                 0
Trade ID            0
Timestamp           0
dtype: int64


Fear & Greed Dataset


timestamp         0
value             0
classification    0
date              0
dtype: int64

In [23]:
print("=" * 70)
print("Duplicate Records")
print("=" * 70)

print("Historical Dataset:", historical_df.duplicated().sum())

print("Fear & Greed Dataset:", sentiment_df.duplicated().sum())

Duplicate Records
Historical Dataset: 0
Fear & Greed Dataset: 0


## Observations

The initial inspection reveals that both datasets are clean and well-structured.

### Historical Trader Dataset

- Contains **211,224** trading records.
- Contains **16** features describing trader activity.
- No missing values were detected.
- No duplicate records were found.
- Includes numerical, categorical, boolean and timestamp-related features.

### Bitcoin Fear & Greed Dataset

- Contains **2,644** daily sentiment records.
- Includes market sentiment classifications and sentiment scores.
- No missing values were detected.
- No duplicate records were found.

### Next Step

Since the datasets contain no missing or duplicate values, the preprocessing stage will mainly focus on:

- Converting timestamps into datetime format
- Standardizing date columns
- Creating new analytical features
- Preparing both datasets for merging

In [25]:
print("=" * 70)
print("Historical Dataset Summary")
print("=" * 70)

display(historical_df.describe(include="all"))

print("=" * 70)
print("Fear & Greed Dataset Summary")
print("=" * 70)

display(sentiment_df.describe(include="all"))

Historical Dataset Summary


,Account,Coin,Execution Price,Size Tokens,Size USD,Side,Timestamp IST,Start Position,Direction,Closed PnL,Transaction Hash,Order ID,Crossed,Fee,Trade ID,Timestamp
count,211224,211224,211224.00,211224.00,211224.00,211224,211224,211224.00,211224,211224.00,211224,211224.00,211224,211224.00,211224.00,211224.00
unique,32,246,NaN,NaN,NaN,2,27977,NaN,12,NaN,101184,NaN,2,NaN,NaN,NaN
top,0xbee1707d6b44d4d52bfe19e41f8a828645437aab,HYPE,NaN,NaN,NaN,SELL,14-02-2025 00:31,NaN,Open Long,NaN,0x00000000000000000000000000000000000000000000...,NaN,True,NaN,NaN,NaN
freq,40184,68005,NaN,NaN,NaN,108528,441,NaN,49895,NaN,9032,NaN,128403,NaN,NaN,NaN
mean,NaN,NaN,11414.72,4623.36,5639.45,NaN,NaN,-29946.25,NaN,48.75,NaN,69653876008.97,NaN,1.16,562854854500788.50,1737744290421.54
std,NaN,NaN,29447.65,104272.89,36575.14,NaN,NaN,673807.42,NaN,919.16,NaN,18357525271.93,NaN,6.76,325756470346523.38,8689920301.59
min,NaN,NaN,0.00,0.00,0.00,NaN,NaN,-14334629.00,NaN,-117990.10,NaN,173271100.00,NaN,-1.18,0.00,1680000000000.00
25%,NaN,NaN,4.85,2.94,193.79,NaN,NaN,-376.23,NaN,0.00,NaN,59838527992.75,NaN,0.02,281000000000000.00,1740000000000.00
50%,NaN,NaN,18.28,32.00,597.05,NaN,NaN,84.73,NaN,0.00,NaN,74429390066.00,NaN,0.09,562000000000000.00,1740000000000.00
75%,NaN,NaN,101.58,187.90,2058.96,NaN,NaN,9337.28,NaN,5.79,NaN,83355430544.00,NaN,0.39,846000000000000.00,1740000000000.00


Fear & Greed Dataset Summary


,timestamp,value,classification,date
count,2644.00,2644.00,2644,2644
unique,NaN,NaN,5,2644
top,NaN,NaN,Fear,2018-02-01
freq,NaN,NaN,781,1
mean,1631898885.02,46.98,NaN,NaN
std,65979674.32,21.83,NaN,NaN
min,1517463000.00,5.00,NaN,NaN
25%,1574811000.00,28.00,NaN,NaN
50%,1631899800.00,46.00,NaN,NaN
75%,1688988600.00,66.00,NaN,NaN


In [27]:
summary = pd.DataFrame({
    "Dataset": ["Historical Trader Data", "Fear & Greed Index"],
    "Rows": [historical_df.shape[0], sentiment_df.shape[0]],
    "Columns": [historical_df.shape[1], sentiment_df.shape[1]],
    "Missing Values": [
        historical_df.isnull().sum().sum(),
        sentiment_df.isnull().sum().sum()
    ],
    "Duplicate Rows": [
        historical_df.duplicated().sum(),
        sentiment_df.duplicated().sum()
    ]
})

display(summary)

,Dataset,Rows,Columns,Missing Values,Duplicate Rows
0,Historical Trader Data,211224,16,0,0
1,Fear & Greed Index,2644,4,0,0


# Initial Findings

The preliminary inspection provides an overview of the datasets before preprocessing.

Key observations to verify:

- Number of records available in each dataset.
- Presence of missing values.
- Presence of duplicate records.
- Data types that may require conversion.
- Features available for further analysis.

These observations will guide the data cleaning and feature engineering process in the next section.

# 5. Data Cleaning & Preprocessing

The datasets are free from missing values and duplicate records. Therefore, preprocessing focuses on improving data consistency and preparing the datasets for analysis.

The preprocessing steps include:

- Converting timestamps into datetime format
- Standardizing date formats
- Verifying data types
- Creating additional features for analysis

In [32]:
# Create copies of original datasets

historical_clean = historical_df.copy()
sentiment_clean = sentiment_df.copy()

In [34]:
# Convert date column

sentiment_clean["date"] = pd.to_datetime(sentiment_clean["date"])

print("Fear & Greed date converted successfully.")

Fear & Greed date converted successfully.


In [36]:
historical_clean["Timestamp"] = pd.to_datetime(
    historical_clean["Timestamp"],
    unit="ms"
)

In [42]:
historical_clean["Timestamp IST"] = pd.to_datetime(
    historical_clean["Timestamp IST"],
    dayfirst=True
)

In [44]:
historical_clean["Timestamp"].head()

0   2024-10-27 03:33:20
1   2024-10-27 03:33:20
2   2024-10-27 03:33:20
3   2024-10-27 03:33:20
4   2024-10-27 03:33:20
Name: Timestamp, dtype: datetime64[ns]

In [46]:
historical_clean[["Timestamp", "Timestamp IST"]].head()

,Timestamp,Timestamp IST
0,2024-10-27 03:33:20,2024-12-02 22:50:00
1,2024-10-27 03:33:20,2024-12-02 22:50:00
2,2024-10-27 03:33:20,2024-12-02 22:50:00
3,2024-10-27 03:33:20,2024-12-02 22:50:00
4,2024-10-27 03:33:20,2024-12-02 22:50:00


In [48]:
sentiment_clean["date"].head()

0   2018-02-01
1   2018-02-02
2   2018-02-03
3   2018-02-04
4   2018-02-05
Name: date, dtype: datetime64[ns]

## Feature Engineering

To prepare the datasets for analysis, we create additional features that simplify merging and enable time-based insights.

The new features include:

- Trading Date
- Trading Hour
- Trading Day
- Market Sentiment Date

In [51]:
# Extract only the date from Timestamp IST

historical_clean["Trade Date"] = historical_clean["Timestamp IST"].dt.date

historical_clean[["Timestamp IST", "Trade Date"]].head()

,Timestamp IST,Trade Date
0,2024-12-02 22:50:00,2024-12-02
1,2024-12-02 22:50:00,2024-12-02
2,2024-12-02 22:50:00,2024-12-02
3,2024-12-02 22:50:00,2024-12-02
4,2024-12-02 22:50:00,2024-12-02


In [53]:
historical_clean["Trade Hour"] = historical_clean["Timestamp IST"].dt.hour

historical_clean["Trade Hour"].head()

0    22
1    22
2    22
3    22
4    22
Name: Trade Hour, dtype: int32

In [55]:
historical_clean["Day of Week"] = historical_clean["Timestamp IST"].dt.day_name()

historical_clean["Day of Week"].head()

0    Monday
1    Monday
2    Monday
3    Monday
4    Monday
Name: Day of Week, dtype: object

In [57]:
sentiment_clean["Trade Date"] = sentiment_clean["date"].dt.date

sentiment_clean[["date", "Trade Date"]].head()

,date,Trade Date
0,2018-02-01,2018-02-01
1,2018-02-02,2018-02-02
2,2018-02-03,2018-02-03
3,2018-02-04,2018-02-04
4,2018-02-05,2018-02-05


In [59]:
print(historical_clean[["Trade Date", "Trade Hour", "Day of Week"]].dtypes)

print()

print(sentiment_clean[["Trade Date"]].dtypes)

Trade Date     object
Trade Hour      int32
Day of Week    object
dtype: object

Trade Date    object
dtype: object


## Date Range Validation

Before merging the datasets, it is important to verify that both datasets cover overlapping time periods.

This validation ensures that each trade can be matched with the corresponding Bitcoin market sentiment based on the trading date.

In [62]:
print("=" * 70)
print("Historical Trader Dataset Date Range")
print("=" * 70)

print("Start Date :", historical_clean["Trade Date"].min())
print("End Date   :", historical_clean["Trade Date"].max())

print()

print("=" * 70)
print("Fear & Greed Dataset Date Range")
print("=" * 70)

print("Start Date :", sentiment_clean["Trade Date"].min())
print("End Date   :", sentiment_clean["Trade Date"].max())

Historical Trader Dataset Date Range
Start Date : 2023-05-01
End Date   : 2025-05-01

Fear & Greed Dataset Date Range
Start Date : 2018-02-01
End Date   : 2025-05-02


In [64]:
common_dates = set(historical_clean["Trade Date"]).intersection(
    set(sentiment_clean["Trade Date"])
)

print(f"Common Dates Available: {len(common_dates)}")

Common Dates Available: 479


# 6. Merge Datasets

The historical trading data is merged with the Bitcoin Fear & Greed Index using the common **Trade Date**.

This allows every trade to be associated with the market sentiment on the day it occurred.

In [69]:
merged_df = historical_clean.merge(
    sentiment_clean[
        ["Trade Date", "classification", "value"]
    ],
    on="Trade Date",
    how="left"
)

merged_df.head()

,Account,Coin,Execution Price,Size Tokens,Size USD,Side,Timestamp IST,Start Position,Direction,Closed PnL,Transaction Hash,Order ID,Crossed,Fee,Trade ID,Timestamp,Trade Date,Trade Hour,Day of Week,classification,value
0,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.98,986.87,7872.16,BUY,2024-12-02 22:50:00,0.00,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.35,895000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
1,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.98,16.00,127.68,BUY,2024-12-02 22:50:00,986.52,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.01,443000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
2,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,144.09,1150.63,BUY,2024-12-02 22:50:00,1002.52,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.05,660000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
3,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,142.98,1142.04,BUY,2024-12-02 22:50:00,1146.56,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.05,1080000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
4,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,8.73,69.75,BUY,2024-12-02 22:50:00,1289.49,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.00,1050000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00


In [71]:
merged_df.rename(
    columns={
        "classification": "Market Sentiment",
        "value": "Fear & Greed Value"
    },
    inplace=True
)

merged_df.head()

,Account,Coin,Execution Price,Size Tokens,Size USD,Side,Timestamp IST,Start Position,Direction,Closed PnL,Transaction Hash,Order ID,Crossed,Fee,Trade ID,Timestamp,Trade Date,Trade Hour,Day of Week,Market Sentiment,Fear & Greed Value
0,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.98,986.87,7872.16,BUY,2024-12-02 22:50:00,0.00,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.35,895000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
1,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.98,16.00,127.68,BUY,2024-12-02 22:50:00,986.52,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.01,443000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
2,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,144.09,1150.63,BUY,2024-12-02 22:50:00,1002.52,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.05,660000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
3,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,142.98,1142.04,BUY,2024-12-02 22:50:00,1146.56,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.05,1080000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00
4,0xae5eacaf9c6b9111fd53034a602c192a04e082ed,@107,7.99,8.73,69.75,BUY,2024-12-02 22:50:00,1289.49,Buy,0.00,0xec09451986a1874e3a980418412fcd0201f500c95bac...,52017706630,True,0.00,1050000000000000.00,2024-10-27 03:33:20,2024-12-02,22,Monday,Extreme Greed,80.00


In [73]:
print("=" * 70)
print("Merged Dataset")
print("=" * 70)

print("Rows    :", merged_df.shape[0])
print("Columns :", merged_df.shape[1])

print()

print("Missing Market Sentiment:",
      merged_df["Market Sentiment"].isnull().sum())

Merged Dataset
Rows    : 211224
Columns : 21

Missing Market Sentiment: 6


In [75]:
merged_df.to_csv(
    "../data/processed/merged_dataset.csv",
    index=False
)

print("Merged dataset saved successfully!")

Merged dataset saved successfully!


## Merge Validation

The datasets were successfully merged using the common **Trade Date**.

Out of **211,224** trading records, only **6** records could not be matched with a market sentiment value.

These unmatched records will be inspected before proceeding with the analysis.

In [78]:
missing_sentiment = merged_df[
    merged_df["Market Sentiment"].isna()
]

missing_sentiment

,Account,Coin,Execution Price,Size Tokens,Size USD,Side,Timestamp IST,Start Position,Direction,Closed PnL,Transaction Hash,Order ID,Crossed,Fee,Trade ID,Timestamp,Trade Date,Trade Hour,Day of Week,Market Sentiment,Fear & Greed Value
727,0x72c6a4624e1dffa724e6d00d64ceae698af892a0,EIGEN,2.83,10000.00,28330.00,BUY,2024-10-26 04:44:00,-32797.51,Close Short,11655.00,0x4da9426d2bf6ec027d8404152f9bad02010700c745eb...,40671411386,False,2.72,83500000000000.00,2024-10-27 03:33:20,2024-10-26,4,Saturday,NaN,NaN
728,0x72c6a4624e1dffa724e6d00d64ceae698af892a0,EIGEN,2.71,10797.51,29272.05,BUY,2024-10-26 04:52:00,-22797.51,Close Short,13901.79,0x66588e180b942ff0e4d104152fb6f20000ac4ae48a39...,40671438761,False,2.81,323000000000000.00,2024-10-27 03:33:20,2024-10-26,4,Saturday,NaN,NaN
729,0x72c6a4624e1dffa724e6d00d64ceae698af892a0,EIGEN,2.59,6908.33,17884.98,BUY,2024-10-26 04:52:00,-12000.00,Close Short,9737.98,0x66588e180b942ff0e4d104152fb6f20000ac4ae48a39...,40671466136,False,1.72,1070000000000000.00,2024-10-27 03:33:20,2024-10-26,4,Saturday,NaN,NaN
730,0x72c6a4624e1dffa724e6d00d64ceae698af892a0,EIGEN,2.59,529.39,1370.54,BUY,2024-10-26 04:52:00,-5091.67,Close Short,746.23,0x66588e180b942ff0e4d104152fb6f20000ac4ae48a39...,40671466136,False,0.13,877000000000000.00,2024-10-27 03:33:20,2024-10-26,4,Saturday,NaN,NaN
731,0x72c6a4624e1dffa724e6d00d64ceae698af892a0,EIGEN,2.59,38.65,100.06,BUY,2024-10-26 04:52:00,-4562.28,Close Short,54.48,0x66588e180b942ff0e4d104152fb6f20000ac4ae48a39...,40671466136,False,0.01,982000000000000.00,2024-10-27 03:33:20,2024-10-26,4,Saturday,NaN,NaN
732,0x72c6a4624e1dffa724e6d00d64ceae698af892a0,EIGEN,2.59,4523.63,11711.23,BUY,2024-10-26 04:53:00,-4523.63,Close Short,6376.51,0x39ce7efcc1773fd0eec704152fbcb4010100c8beadd4...,40671466136,False,1.12,936000000000000.00,2024-10-27 03:33:20,2024-10-26,4,Saturday,NaN,NaN


In [80]:
missing_sentiment[[
    "Trade Date",
    "Timestamp IST",
    "Coin"
]]

,Trade Date,Timestamp IST,Coin
727,2024-10-26,2024-10-26 04:44:00,EIGEN
728,2024-10-26,2024-10-26 04:52:00,EIGEN
729,2024-10-26,2024-10-26 04:52:00,EIGEN
730,2024-10-26,2024-10-26 04:52:00,EIGEN
731,2024-10-26,2024-10-26 04:52:00,EIGEN
732,2024-10-26,2024-10-26 04:53:00,EIGEN


In [82]:
historical_clean.to_csv(
    "../data/processed/historical_clean.csv",
    index=False
)

sentiment_clean.to_csv(
    "../data/processed/sentiment_clean.csv",
    index=False
)

merged_df.to_csv(
    "../data/processed/merged_dataset.csv",
    index=False
)